Add MaxTxSize check to ValidateExpressLaneTx()

KolbyML opened the pull request:
resolves NIT-4157

The premise is we should check if a transaction is valid, before we send it when possible.

We already check the MaxTxDataSize in `ValidateTransaction()`, so we should do it for `ValidateExpressLaneTx()` as well.

I am using uint64, as I don't think we should use `int` as it
- varies in size depending on the platform
- the parameter can't be negative

The sequencer takes  length of `msg.Transaction.MarshalBinary()` for its internal size check, I am surprised it doesn't use `.Size()`,  that is why I must do the same here.

ganeshvanahalli reviewed (commented):
We do already have MaxTxSize check in the sequencer right? https://github.com/OffchainLabs/nitro/blob/6837a8fd5d80b6eee74a5b6ffec25fb281225bd3/execution/gethexec/sequencer.go#L1295-L1299

KolbyML commented:
> We do already have MaxTxSize check in the sequencer right?
> 
> https://github.com/OffchainLabs/nitro/blob/6837a8fd5d80b6eee74a5b6ffec25fb281225bd3/execution/gethexec/sequencer.go#L1295-L1299

We do check for this in the sequencer, we should fail transactions before they reach the sequencer when possible.

KolbyML commented:
@joshuacolvin0 helped show me Transaction.Size() and len(MarshalBinary) are the same thing. I updated the sequencer to use .Size(), along with the validation check I added.
